server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java

```diff
@@ -80,9 +79,47 @@
      * @throws MessagingException when the message is not clonable
      */
     public static MailImpl duplicate(Mail mail) throws MessagingException {
-        return new MailImpl(mail, newName(mail));
+        return new MailImpl(mail, deriveNewName(mail.getName()));
     }
 
+    private static final java.util.Random random = new java.util.Random(); // Used
```

Comment:
FQDN + weiiird comment

Reply:
I changed it after covering with test, that's own it's supposed to be done, right ?